go/datas/http_chunk_store.go

```diff
@@ -355,53 +358,37 @@ func resBodyReader(res *http.Response) (reader io.ReadCloser) {
 func (hcs *httpChunkStore) Put(c chunks.Chunk) {
 	hcs.cacheMu.RLock()
 	defer hcs.cacheMu.RUnlock()
+	select {
+	case <-hcs.finishedChan:
+		d.Panic("Tried to Put %s into closed ChunkStore", c.Hash())
+	default:
+	}
 	hcs.unwrittenPuts.Insert(c)
 }
 
-func (hcs *httpChunkStore) sendWriteRequests() {
-	hcs.rateLimit <- struct{}{}
-	defer func() { <-hcs.rateLimit }()
-
-	hcs.cacheMu.Lock()
-	defer func() {
-		hcs.cacheMu.Unlock()
-	}()
-
-	count := hcs.unwrittenPuts.Count()
-	if count == 0 {
-		return
-	}
-	defer func() {
-		hcs.unwrittenPuts.Destroy()
-		hcs.unwrittenPuts = nbs.NewCache()
-	}()
-
-	verbose.Log("Sending %d chunks", count)
+func sendWriteRequest(u url.URL, auth, vers string, p *nbs.NomsBlockCache, cli httpDoer) {
 	chunkChan := make(chan *chunks.Chunk, 1024)
 	go func() {
-		hcs.unwrittenPuts.ExtractChunks(chunkChan)
+		p.ExtractChunks(chunkChan)
 		close(chunkChan)
 	}()
 
 	body := buildWriteValueRequest(chunkChan)
-	url := *hcs.host
-	url.Path = httprouter.CleanPath(hcs.host.Path + constants.WriteValuePath)
 	// TODO: Make this accept snappy encoding
-	req := newRequest("POST", hcs.auth, url.String(), body, http.Header{
+	req := newRequest("POST", auth, u.String(), body, http.Header{
```

Comment:
remove TODO above? don't see much point in accepting snappy encoding response to this request.